perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -54,6 +80,107 @@
                   'num_machines': 1}
 
 
+JOB_FILE_TEMPLATE = """
```

Comment:
@stfeng @yuyantingzero 

I've been thinking about this, and I'd like to propose that we handle the change by generalizing the template-based code so it can generate something similar to the old fio.job file, and then making that the default. My vision for the new code is that it lets you essentially take the old job file and vary some parameters to see what happens. Right now it only lets you vary queue depth, but we could easily make it vary more parameters without changing the interface.

The parts of the old job file that will be difficult to generate exactly with my new code are the fact that the jobs use different sizes, the fact that the sizes depend on $mb_memory, and the fact that one job has end_fsync=1 and none of the others do. If you two think those are important things to preserve, then I can make it use the old job file by default. But I think it would be much cleaner to have the default be generating a test that happens to look like the old fio.job file, while giving the user the option to still run exactly the old fio.job file.

Also, since we're hoping to release 1.0 soon, I think we should be asking not just will this change break someone's automation, but also, what are the defaults that we want to keep for a long time? If we're going to change, I'd much rather do it before 1.0 than after it.